lib/elasticsearch.js

```diff
@@ -0,0 +1,540 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+const AWSESConnection = require('http-aws-es');
+
+const UUID = require('uuid/v4');
+
+const elasticsearch = require('@elastic/elasticsearch');
+
+const ElasticSearchFilters = require('./elasticsearch-filters');
+const ElasticSearchError = require('./elasticsearch-error');
+
+const DEFAULT_LIMIT = 500;
+const ES_DEFAULT_HOST = 'http://localhost';
+
+/**
+ * @class ElasticSearch
+ * @classdesc Elasticsearch driver module
+*/
+class ElasticSearch {
+
+	constructor(config = {}) {
+		if(typeof config !== 'object' || Array.isArray(config))
+			throw new ElasticSearchError('Invalid config', ElasticSearchError.codes.INVALID_CONFIG);
+
+		this.config = {
+			protocol: config.protocol || '',
+			host: config.host || ES_DEFAULT_HOST,
+			port: config.port || '',
+			user: config.user || '',
+			password: config.password || '',
+			limit: config.limit || DEFAULT_LIMIT,
+			awsCredentials: config.awsCredentials || false
+		};
+
+		this.config.host = this.config.host.replace(this.config.protocol, '');
+	}
+
+	/**
+	 * Elasticsearch connection URL prefix
+	 * @returns {String} Elasticsearch URL prefix
+	*/
+	get userPrefix() {
+		return this.config.user ? `${this.config.user}:${this.config.password}@` : '';
+	}
+
+	/**
+	 * Checks the connection to the Elasticsearch Client
+	 * @returns {elasticsearch.Client} Elasticsearch client with the connection
+	 * @throws if the connection is not successfull
+	*/
+	getClient() {
+		if(!this.client) {
+			try {
+				this.client = new elasticsearch.Client({
+					node: `${this.config.protocol}${this.userPrefix}${this.config.host}${this.config.port ? `:${this.config.port}` : ''}`,
+					...this.config.awsCredentials ? { connectionClass: AWSESConnection, amazonES: { credentials: new AWS.EnvironmentCredentials('AWS') } } : {}
```

Comment:
Puede ser que estuviera bueno que esta parte (la de agregar o no las credenciales de AWS) sean una función aparte para agregar legibilidad.
En esta linea 
```suggestion
					...this._insertAWScredentials(this.config)
```
La función:
```suggestion
	_insertAWScredentials(config) {
		if(!this.config.awsCredentials) 
			return {};
		return { 
			connectionClass: AWSESConnection, 
			amazonES: { credentials: new AWS.EnvironmentCredentials('AWS') } 
		}
	}
```